velox/functions/sparksql/tests/SplitFunctionsTest.cpp

```diff
@@ -105,5 +106,76 @@ TEST_F(SplitTest, longStrings) {
       {{{"abcdefghijklkmnopqrstuvwxyz"}}});
 }
 
+class StrToMapTest : public SparkFunctionBaseTest {
+ protected:
+  void testStrToMap(
+      const std::vector<StringView>& parameters,
+      const std::vector<std::pair<StringView, std::optional<StringView>>>&
+          expect) {
+    std::string expr;
+    switch (parameters.size()) {
+      case 1:
+        expr = "str_to_map(c0)";
+        break;
+      case 2:
+        expr = "str_to_map(c0, c1)";
+        break;
+      case 3:
+        expr = "str_to_map(c0, c1, c2)";
+        break;
+      default:
+        VELOX_FAIL("Unsupported arguments size");
+    }
+    std::vector<VectorPtr> row;
+    row.reserve(parameters.size());
+    for (auto parameter : parameters) {
+      row.emplace_back(makeFlatVector<StringView>({parameter}));
+    }
+    auto result = evaluate<MapVector>(expr, makeRowVector(row));
+    auto expected = makeMapVector<StringView, StringView>({expect});
+    ::facebook::velox::test::assertEqualVectors(result, expected);
+  }
+};
+
+TEST_F(StrToMapTest, Basics) {
+  // clang-format off
+  auto table = std::vector<std::tuple<
+    std::vector<StringView>,
+    std::vector<std::pair<StringView, std::optional<StringView>>>>>{
+    {
+            {"a:1,b:2,c:3"},
```

Comment:
Can you test this delimiter?
```
spark-sql> SELECT str_to_map('a::1,b::2,c::3', ',', '::');
{"a":"1","b":"2","c":"3"}
```

Reply:
And the test:
```
spark-sql> SELECT str_to_map('a::1,b::2,c::3', ',', 'c');
{"":"::3","a::1":null,"b::2":null}

```

Reply:
Just added the 2nd test case. For the first one, as this PR just implements basic single-character delimiter, let's cover multi-character in separate PR. Thanks!